rastervision_core/rastervision/core/rv_pipeline/semantic_segmentation_config.py

```diff
@@ -61,6 +61,17 @@ class SemanticSegmentationConfig(RVPipelineConfig):
     chip_options: SemanticSegmentationChipOptions = SemanticSegmentationChipOptions(
     )
 
+    img_channels: int = Field(
```

Comment:
It seems like `img_channels` and `img_format` could and should be determined from the `channel_order` in `RasterSourceConfig`. If `len(channel_order) == 3`, then set the format to png. If > 3, then use npy. This will help users avoid invalid configuration. I think it makes sense to still have the ability to manually override things, but the default should be to avoid needing extra configuration. 

More concretely, I'm suggesting that we make the `img_channels` and `img_format` Optional with default value None, and then make the `update` method fill in those values based on the length of the `channel_order` array in the `RasterSourceConfig` in the first scene.

Also, it would be good to validate `img_channels` and `img_format` so that they are consistent with one another. For instance, `img_channels=4` and `img_format='png'` would be invalid.

Reply:
That makes sense. I have changed it so that `DatasetConfig` sets a `img_channels` property on itself based on the first training scene. And since this assumes that `channel_order` for all scenes must be the same, I have added a validation check for that.

`img_format` is also now initialized based on `img_channels`.